### ISSUE TITLE
minor comments

### ISSUE THREAD

**Issue by vincentarelbundock** [stata2r/stata2r.github.io#8]
This is amazing work! Wow! I'm sure it will be super useful to people.

Here are a few comments I jotted down while reading. None of them are important, and all could easily be ignored. 

My main comment is that I feel it can often be intimidating/confusing for begginers when we show them multiple ways of doing a single task. If the goal is to turn them into `data.table` expert, I can see why it would make sense to show them 3 ways of selecting columns. But if the goal is to help people transition from Stata to R, then showing them one good way is probably sufficient (and arguably better).

But again, feel free to ignore. The page is amazing as it is.

+ [x] Data I/O save to binary file using `saveRDS()`. If the cheat sheet is about *interoperability*, this is not needed. But if the cheat sheet aims to help people *transition* from `Stata` to `R`, then it is useful to know the native binary format.
+ [x] `setnames(dat, gsub('arr_', 'arrival_', names(dat)))` would shorter and keep the syntax consistent with your other examples of renaming (only one obvious way, etc.).
+ [x] Be more opinionated about a preferred way. One obstacle to learning is when we see multiple alternatives that achieve the same goal. ex:
    - 3 ways of selecting columns
    - 2 ways of dropping columns
    - 2 ways to do rowwise calculations
    - "Data-tably" merge strategy is probably hard for newcomers. Is there a performance benefit to using `setkey` before merge? If not, why show it?
+ [x] Linting, ex: spaces after commas and around infix operators
+ [x] Do not rely on argument order. Name them explicitly. ex:
    - `na.omit(dat, cols = c('air_time', 'dest'))`
+ [x] Why show both `let` and `:=`? BTW, when I learned about `let` I was kind of hoping it would allow `let(mpg2 = mpg*2, mpg4 = mpg2*2)`. :-\
+ [x] "Unlike Stata, multiple datasets can exist in memory and R's collapse operations won't overwrite your original dataset." I thought I heard Stata people say this was no longer a limitation. 
+ [x] Count rows by group: `dat[, .N, by = "id"]`
+ [x] `feols(y ~ x, dat, vcov = 'NW', panel.id = ~unit + time)` and remove the comment
+ [x] #17

```{r}
dat[, dist_sq := distance^2][
    , dist_cu := distance*dist_sq)]
```

You have probably seen this, but if not there might be interesting things here: https://atrebas.github.io/post/2019-03-03-datatable-dplyr/

**Comment by NickCH-K** [comment]
Thanks for the comments! Some notes on these:

- `saveRDS()` - we had a discussion about this in another issue thread (I forget where) and ended up leaving it out in favor of CSV/parquet
- Tend to agree on the multiple-ways problem. We should probably trim back to one method for most if not all of these
- `let()` is only in the dev version so we can't really switch to it entirely, but it is nice to see. Then again if we trim back to one approach each it will likely be dropped
- multiple data sets in Stata is possible now but kind of weird and nonstandard. But we should probably edit this to be more accurate

**Comment by NickCH-K** [comment]
Any opinions on culling the multiple-ways-of-doing @grantmcdermott @kylebutts ? Seems to me like these are the culprits:

- Multiple ways of creating variables, i.e. four variants beginning `dat[, c('dist_sq', 'dist_cu') := .(distance^2, distance^3)]`. IMO the version I just quoted I'd prefer since it translates easily to `lapply()/.SD` applications. Perhaps to be updated with `let()` but only once that's on CRAN.
- Custom row calculations: we have both `mapply` and `by=1:nrow(dat)`. I'm torn on which to do - `mapply` requires you to start keeping track of which `apply` function you're using which is an R turnoff. But I think `by=1:nrow(dat)` is slower (right?)
- Collapse with `.SD` - torn on this one, I think the multiple ways of picking `.SDcols` is handy.
- Collapse by group with `.SD` - we have both `.SDcols=c('dep_delay','arr_delay','air_time','distance')` and `.SDcols = c(4,5,9,10)`, I'd stick with the first one.
- I think we could drop the `dcast()` collapse application. Like Vincent said, this is advanced stuff, not beginner
- Also could see the benefit of dropping the `setkey()` and `D1[D2, on=]` merges (perhaps linking out to a page about setting keys generally, and a page about `D1[D2, on=]` merging, since there are speed gains, especially with keys).
- We could probably pare back the five merge varieties to to just a note that `keep()` and `all.x, all.y` control what you keep. Maybe two **data.table** merges, one with `by` and one with `by.x` and `by.y`.

That's how I would choose to resolve these if we decide to limit the number of ways to do something.

**Comment by grantmcdermott** [comment]
Hey guys, sorry I'm late responding. Still catching up after my vacation.

Couple thoughts: 

- First: Thanks Vincent, these are super helpful!
- There's definitely a good argument to be made for sticking to one, opinionated way of doing things and it's something I try to do normally. But in this particular case, I lean towards keeping multiple options (or, at least most of them). Based on my previous experience helping Stata users translate their code to data.table, there are pretty disparate preferences. Some really don't like wrapping variable names in quotes, others don't the `.()` NSE notation, etc. So I'd prefer to keep our bases at least somewhat covered, as long as these cases are clearly demarcated. 
- Agree with all the other points. I'll add two quick comments, which relate to linting and indentation: 
- We don't have a lexer for the Stata code at the moment. It would be great if we could do this. (Kevin sorted this out for us at LOST so we can always ping him. I know that Julian Reif as a long-pending PR at the main rouge repo for this too.) 
- Comments, spacing and indentation for the Stata code is also a little bit messy b/c its parser doesn't accept `//` and `///` unless fed through a do-file. Should we just say screw it and assume readers understand this? It might make the longer commands easier to read and then we could split the equivalent data.table commands over multiple lines too.

**Comment by grantmcdermott** [comment]
Closing as these look pretty much taken care of now, esp. with the new vuepress build. Thanks again for the suggestions, @vincentarelbundock!
